src/nodes/kvs/store.rs

```diff
@@ -0,0 +1,184 @@
+use crate::{
+    lattice::{
+        causal::SingleKeyCausalLattice,
+        last_writer_wins::{Timestamp, TimestampValuePair},
+        CounterLattice, LastWriterWinsLattice, Lattice, MapLattice, MaxLattice, SetLattice,
+    },
+    messages::{
+        request::{ClientKeyOperation, InnerKeyOperation},
+        response::ClientResponseValue,
+    },
+    store::LatticeValue,
+    AnnaError, Key,
+};
+
+use super::KvsNode;
+
+impl KvsNode {
+    pub(crate) fn key_operation_handler(
+        &mut self,
+        operation: ClientKeyOperation,
+        timestamp: chrono::DateTime<chrono::Utc>,
+    ) -> Result<Option<ClientResponseValue>, AnnaError> {
+        use std::collections::hash_map::Entry;
+
+        match operation {
+            ClientKeyOperation::Get(key) => {
+                if let Some(value) = self.kvs.get(&key.into()) {
+                    Ok(Some(value.clone().into()))
+                } else {
+                    Err(AnnaError::KeyDoesNotExist)
+                }
+            }
+            ClientKeyOperation::Put(key, value) => {
+                let key = Key::Client(key);
+                self.kvs.put(
+                    key.clone(),
+                    LatticeValue::Lww(LastWriterWinsLattice::new(TimestampValuePair::new(
+                        Timestamp(timestamp),
+                        value,
+                    ))),
+                )?;
+                self.local_changeset.insert(key);
+                Ok(None)
+            }
+            ClientKeyOperation::SetAdd(key, value) => {
+                let key = Key::Client(key);
+                let value = SetLattice::new(value);
+
+                match self.kvs.entry(key.clone()) {
+                    Entry::Vacant(entry) => {
+                        entry.insert(LatticeValue::SingleCausal(
+                            SingleKeyCausalLattice::create_with_default_clock(value),
```

Comment:
This creates an empty vector clock, but if the key already exists, we insert a `format!("{}/{}", self.node_id` entry into the vector clock (below). Is this intended?